packages/bitcore-wallet-service/src/lib/storage.ts

```diff
@@ -2520,6 +2530,218 @@ export class Storage {
     );
   }
 
+  fetchAllLogDevice(opts, cb) {
+    let isActive = opts?.isActive;
+    const filter: { countNumber?: object } = {};
+
+    if (isActive) filter.countNumber = { $gt: 0 };
+    this.db
+      .collection(collections.LOG_DEVICE)
+      .find(filter)
+      .toArray((err, listLogDevice) => {
+        if (err) return cb(err);
+        else {
+          const devices = _.map([].concat(listLogDevice), r => {
+            return LogDevice.fromObj(r);
+          });
+          return cb(null, devices);
+        }
+      });
+  }
+
+  getLogDeviceById(deviceId, cb) {
+    this.db.collection(collections.LOG_DEVICE).findOne(
+      {
+        deviceId
+      },
+      (err, result) => {
+        if (err || !result) return cb(err);
+        return cb(null, LogDevice.fromObj(result));
+      }
+    );
+  }
+
+  storeLogDevice(deviceInfo, cb) {
+    // This should only happens in certain tests.
+    if (!this.db) {
+      logger.warn('Trying to store a log device with close DB', deviceInfo);
+      return;
+    }
+
+    this.db.collection(collections.LOG_DEVICE).insertOne(
+      deviceInfo,
+      {
+        w: 1
+      },
+      (err, result) => {
+        if (err) return cb(err);
+        if (!result) return cb();
+
+        return cb(null, result.ops[0]);
+      }
+    );
+  }
+
+  updateLogDevice(device, cb) {
+    this.db.collection(collections.LOG_DEVICE).replaceOne(
+      {
+        deviceId: device.deviceId
+      },
+      device.toObject(),
+      {
+        w: 1,
+        upsert: true
+      },
+      (err, result) => {
+        if (err) return cb(err);
+        if (!result) return cb();
+
+        return cb(null, result.ops[0]);
+      }
+    );
+  }
+
+  resetCountNumberLogDevice(cb) {
+    this.db.collection(collections.LOG_DEVICE).updateMany(
+      {
+        countNumber: { $gt: 0 }
+      },
+      {
+        $set: {
+          countNumber: 0
+        }
+      },
+      {
+        upsert: false
+      },
+      (err, result) => {
+        if (err) return cb(err);
+        if (!result) return cb(new Error('Can not update daily limit for coin config'));
+
+        return cb(null, result);
+      }
+    );
+  }
+
+  fetchAllAppreciation(opts, cb) {
```

Comment:
Do we really need to allow to get all appreciation? If not maybe we can make the filter deviceId mandatory?

Reply:
I will add TODO in case to consider in the future can change to reasonable